prediction_market_agent/agents/utils.py

```diff
@@ -58,8 +57,9 @@ def market_is_saturated(market: AgentMarket) -> bool:
 def _summarize_learnings(
     memories: list[str],
     prompt_template: PromptTemplate,
-    model: str = DEFAULT_OPENAI_MODEL,
+    model: str | None = None,
 ) -> str:
+    model = model if model else DEFAULT_OPENAI_MODEL
```

Comment:
Why?

Reply:
I was using 4o-mini for lower costs. Any reason to NOT have this optionality, which is already part of `_summarize_learnings`?
This does not change current behaviour.

Reply:
That's fine, what I don't understand is the change from

```
model: str = DEFAULT_OPENAI_MODEL,
```

to 

```
    model: str | None = None,
) -> str:
    model = model if model else DEFAULT_OPENAI_MODEL
```

Technically, that's the same thing, just being done in a more complex way.

Previously when model had default argument DEFAULT_OPENAI_MODEL, you could also override it using `model="gpt-4o-mini"`.

Reply:
Sure, why not. Added the default param instead of None.